spring-cloud-aws-sqs/src/main/java/io/awspring/cloud/sqs/ConfigUtils.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright 2013-2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.awspring.cloud.sqs;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Objects;
+import java.util.function.BiConsumer;
+import java.util.function.Consumer;
+import org.springframework.util.CollectionUtils;
+
+/**
+ * Utility class for conditional configurations.
+ *
+ * @author Tomaz Fernandes
+ * @since 3.0
+ */
+public class ConfigUtils {
```

Comment:
same here about readibility

Reply:
Not sure what you mean here as well...

Would rather have tons of `instance of` and `if xxx != null` spread throughout the code?

IMHO this makes the code a lot more expressive and succinct overall, and doesn't have a learning curve to figure out what it does.

Just as other APIs such as `PropertyMapper` - we could simply use `if` conditionals, but the API makes the code more expressive and right to the point.

Of course, in some cases where it seems it doesn't help, I've left the `instance of` and `if xxx != null`.

But that's sure more a matter of personal opinion than a right/wrong issue, I guess.

LMKWYT.